src/openzaak/components/zaken/tests/test_zaak_closed.py

```diff
@@ -124,3 +156,371 @@ def test_reopenzaak_not_allowed(self):
+        )
+        io_url = reverse(io)
+        zio = ZaakInformatieObjectFactory(
+            zaak=self.zaak,
+            informatieobject__latest_version__informatieobjecttype__zaaktypen__zaaktype=self.zaaktype,
+            informatieobject__latest_version__informatieobjecttype__catalogus=self.zaaktype.catalogus,
+        )
+        zio_url = reverse(zio)
+
+        self.assertCreateBlocked(
+            reverse(ZaakInformatieObject),
+            {
+                "zaak": reverse(self.zaak),
+                "informatieobject": f"http://testserver{io_url}",
+            },
+        )
+        self.assertUpdateBlocked(zio_url)
+        self.assertPartialUpdateBlocked(zio_url)
+        self.assertDestroyBlocked(zio_url)
+
+    def test_zaakobjecten(self):
+        self.assertCreateBlocked(
+            reverse(ZaakObject),
+            {
+                "zaak": reverse(self.zaak),
+                "object": "https://example.com",
+                "objectType": "overige",
+                "objectTypeOverige": "website",
+            },
+        )
+
+    def test_zaakeigenschappen(self):
+        zaak_eigenschap = ZaakEigenschapFactory.create(zaak=self.zaak)
+        eigenschap_url = reverse(zaak_eigenschap.eigenschap)
+
+        self.assertCreateBlocked(
+            reverse(ZaakEigenschap, kwargs={"zaak_uuid": self.zaak.uuid}),
+            {
+                "zaak": reverse(self.zaak),
+                "eigenschap": f"http://testserver{eigenschap_url}",
+                "waarde": "123",
+            },
+        )
+
+    def test_klantcontacten(self):
+        url = reverse(KlantContact)
+        data = {
+            "zaak": reverse(self.zaak),
+            "datumtijd": "2020-01-30T15:08:00Z",
+        }
+
+        self.assertCreateBlocked(url, data)
+
+    def test_rollen(self):
+        roltype = RolTypeFactory.create(zaaktype=self.zaaktype)
+        rol = RolFactory.create(zaak=self.zaak, roltype=roltype)
+        rol_url = reverse(rol)
+
+        create_url = reverse(Rol)
+        data = {
+            "zaak": reverse(self.zaak),
+            "roltype": f"http://testserver{reverse(roltype)}",
+            "betrokkeneType": "vestiging",
+            "betrokkene": "https://example.com",
+            "roltoelichting": "foo",
+        }
+
+        self.assertCreateBlocked(create_url, data)
+        self.assertDestroyBlocked(rol_url)
+
+    def test_resultaten(self):
+        resultaat = ResultaatFactory.create(zaak=self.zaak)
+        resultaat_url = reverse(resultaat)
+
+        self.assertUpdateBlocked(resultaat_url)
+        self.assertPartialUpdateBlocked(resultaat_url)
+        self.assertDestroyBlocked(resultaat_url)
+
+        resultaat.delete()
+
+        data = {
+            "zaak": reverse(self.zaak),
+            "resultaattype": f"http://testserver{reverse(resultaat.resultaattype)}",
+        }
+        self.assertCreateBlocked(reverse(Resultaat), data)
+
+    def test_zaakbesluiten(self):
```

Comment:
here the `Besluit` resource is only tested. I think it's nice to have `ZaakBesluit` test also, because in case `besluiten` component is external  the check for closed zaak will be made only in `ZaakBesluit` resource

Reply:
I thought about it too, but saw that we were currently doing it this way in `ZaakViewSet` and wanted to keep the behaviour consistent. It's now raising a 403 (not a 400), since it is also a permission error (you're lacking the overide scope). But I think you can argue for both cases.

Reply:
I tried, but it's pretty much impossible with the signals we have.

In the relation side, `Besluit` is leading and `ZaakBesluit` follows. If the `besluiten` component is external, they have to perform the check that the zaak is not closed.